tests/end-to-end/compass_e2e_test.go

```diff
@@ -1,14 +1,993 @@
 package end_to_end
 
 import (
+	"context"
 	"fmt"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"regexp"
+	"strings"
 	"testing"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	gcli "github.com/machinebox/graphql"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
-func TestApiSpec(t *testing.T) {
+var tc = testContext{graphqlizer: graphqlizer{}, gqlFieldsProvider: gqlFieldsProvider{}, cli: gcli.NewClient(getDirectorURL())}
+
+func TestCreateApplicationWithAllSimpleFieldsProvided(t *testing.T) {
+	// GIVEN
+	ctx := context.Background()
+	in := graphql.ApplicationInput{
+		Name:           "wordpress",
+		Description:    ptrString("my first wordperss application"),
+		HealthCheckURL: ptrString("http://mywordpress.com/health"),
+		Labels: &graphql.Labels{
+			"group": []string{"production", "experimental"},
+		},
+		Annotations: &graphql.Annotations{
+			"createdBy": "admin",
+		},
+	}
+
+	appInputGQL, err := tc.graphqlizer.ApplicationInputToGQL(in)
+	require.NoError(t, err)
+	actualApp := ApplicationExt{}
+	resp := resultMapperFor(&actualApp)
+	// WHEN
+	request := gcli.NewRequest(
+		fmt.Sprintf(`mutation {
```

Comment:
Please define the mutation as a separate variable, it will be easier to read

Reply:
maybe it will be enough to apply different formatting:
```
	request := gcli.NewRequest(
		fmt.Sprintf(`mutation {
			result: createApplication(in: %s) {
					%s
				}
			}`,
			appInputGQL, tc.gqlFieldsProvider.ForApplication()))
```
?

Reply:
I would still avoid nesting, but this is just a suggestion. Skip it for now, we can reorganize this code a bit later.